Jenkinsfile

```diff
@@ -18,315 +18,326 @@ pipeline {
 					}
 				}
 			}
 		}
-		stage('Test-Zero-File-Creation') {
+		stage("Test-Zero-File-Creation") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-zero-file.xml ' +
-								'-m zero_file /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+                    withKubeConfig([credentialsId: "dev-kube-config"]) {
+                        sh("kubectl exec -c nres-pipeline ${podName} -- " +
+                            "/usr/bin/sudo -s -E -u archive /opt/conda/bin/pytest --durations=0 " +
+                            "--junitxml=/nres/code/pytest-zero-file.xml -m zero_file /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-zero-file.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "dev-kube-config"]) {
+						    sh("kubectl cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-zero-file.xml pytest-zero-file.xml")
+						    junit "pytest-zero-file.xml"
 						}
-						junit 'pytest-zero-file.xml'
 					}
 				}
 			}
 		}
-		stage('Test-Science-File-Creation') {
+		stage("Test-Science-File-Creation") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-science-files.xml ' +
-								'-m science_files /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+			        withKubeConfig([credentialsId: "dev-kube-config"]) {
+						sh("kubectl exec -c nres-pipeline ${podName} -- " +
+                            "/usr/bin/sudo -s -E -u archive /opt/conda/bin/pytest --durations=0 " +
+                            "--junitxml=/nres/code/pytest-science-files.xml -m science_files /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-science-files.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "dev-kube-config"]) {
+						    sh("kubectl cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-science-files.xml pytest-science-files.xml")
+						    junit "pytest-science-files.xml"
 						}
-						junit 'pytest-science-files.xml'
 					}
 				}
 				success {
 					script {
-						sh('rancher -c ${RANCHERDEV_CREDS} rm --stop --type stack NRESPipelineTest')
+					    withKubeConfig([credentialsId: "dev-kube-config"]) {
+                            sh("helm delete nres-pipe || true")
```

Comment:
You might want to change this to `helm delete --purge nres-pipe`. Otherwise `helm` will keep the deployment history information so you can "un-delete" each release. (Examine the current history with `helm history nres-pipe`, and we could un-delete with `helm rollback nres-pipe $NUM`.) I doubt we need the ability to un-delete a test stack.